holoviews/core/data/__init__.py

```diff
@@ -188,19 +193,29 @@ def __setstate__(self, state):
 
         super(Dataset, self).__setstate__(state)
 
-    def closest(self, coords):
+
+    def closest(self, coords=[], **kwargs):
         """
         Given single or multiple samples along the first key dimension
         will return the closest actual sample coordinates.
         """
         if self.ndims > 1:
-            NotImplementedError("Closest method currently only "
-                                "implemented for 1D Elements")
+            raise NotImplementedError("Closest method currently only "
+                                      "implemented for 1D Elements")
+
+        if kwargs:
+            dim = self.get_dimension(list(kwargs.keys())[0], strict=True)
+            if len(kwargs) > 1:
+                raise NotImplementedError("Closest method currently only "
+                                          "supports 1D indexes")
+            samples = list(kwargs.values())[0]
+            coords = samples if isinstance(samples, list) else [samples]
```

Comment:
>Might be worth always requiring a list of samples for consistency - it all depends on how often a single sample is needed.

I'll make a comment, this is for ``img.sample(x=0)``, which wasn't supported on Datasets before.